src/test/java/teammates/test/cases/browsertests/InstructorCourseStudentDetailsEditPageUiTest.java

```diff
@@ -134,9 +134,15 @@ public void testEditAction() {
                                       "newemail@gmail.tmt");
         assertEquals("New name", student.name);
         assertEquals("New team", student.team);
-        assertEquals(testData.students.get("registeredStudent").googleId, student.googleId);
         assertEquals("newemail@gmail.tmt", student.email);
         assertEquals("New comments", student.comments);
+        assertTrue(student.googleId.isEmpty()); // Due to Google ID reset
+        
+        // Verify adding the original student again does not overwrite the edited entity
+        BackDoor.createStudent(testData.students.get("registeredStudent"));
+        student = BackDoor.getStudent(testData.courses.get("CCSDEditUiT.CS2104").getId(),
+                                      "newemail@gmail.tmt");
```

Comment:
Not related to this PR: `newemail@gmail.tmt` is used a few times in this test. Should be declared as a local variable. Furthermore, the whole `// Verify data` block should be abstracted as a util method.